### ISSUE TITLE
--watch options breaks socket.io (bad gateway)

### ISSUE THREAD

**Issue by umdstu** [Unitech/pm2#4747]
## What's going wrong?
So this seems identical to #2482 from 4 years ago, but it doesn't seem as though any resolution came from that issue. When using the --watch option, my express/socket.io/nginx/cors application fails to play nicely. All websocket requests from the client yield a 502 Bad Gateway, and server-side there seem to be zero errors. Everything works fine when --watch is not passed as an argument.

## How could we reproduce this issue?
create an 'express.js' app with 'socket.io' support that is running on port 3000 and then proxied via nginx. Create an ecosystem.js file that runs 'npm start' with necessary environment variables set, then run `sudo pm2 start ecosystem.js --watch`.

## Supporting information

$ pm2 report
Unable to copy and paste this text to the issue, but it reports the application as running. What is interesting is that the application appears to be restarting 3-4 seconds because of a detected change a file ./logs/app.log. This was unintentional so i changed the logging to write to /var/logs instead, and then suddenly this issue was resolved.

Bottom line- it appears as though if change detection is continuously restarting your application, it won't ever technically be available to handle requests. Although I resolved this by moving my logging out of the application directory, I wanted to still submit this issue incase others came across it. Additionally i thought it might be worth considering adding an ignore files option in an ecosystem.js configuration for running watch.
